mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java

```diff
@@ -126,12 +126,13 @@ public void storeAttachmentsForMessage(Collection<Attachment> attachments, Messa
     @Override
     public Collection<MessageId> getRelatedMessageIds(AttachmentId attachmentId) throws MailboxException {
         return attachmentMessageIdDAO.getOwnerMessageIds(attachmentId)
-            .join();
+            .collectList()
+            .block();
     }
 
     @Override
     public Collection<Username> getOwners(AttachmentId attachmentId) throws MailboxException {
-        return ownerDAO.retrieveOwners(attachmentId).join().collect(Guavate.toImmutableList());
+        return ownerDAO.retrieveOwners(attachmentId).collect(Guavate.toImmutableList()).block();
```

Comment:
Why line 129 we use `collectList()` and here we use `collect(Guavate.toImmutableList())` ?